lifecycle-operator/controllers/lifecycle/keptnworkloadversion/controller.go

```diff
@@ -225,15 +227,15 @@ func (r *KeptnWorkloadVersionReconciler) doPostDeploymentTaskPhase(ctx context.C
 }
 
 func (r *KeptnWorkloadVersionReconciler) doPostDeploymentEvaluationPhase(ctx context.Context, workloadVersion *klcv1beta1.KeptnWorkloadVersion, ctxWorkloadTrace context.Context) (phase.PhaseResult, error) {
-	if !workloadVersion.IsPostDeploymentEvaluationSucceeded() {
+	if !workloadVersion.IsPostDeploymentEvaluationSucceeded(r.Config.GetBlockDeployment()) {
 		reconcilePostEval := func(phaseCtx context.Context) (apicommon.KeptnState, error) {
 			return r.reconcilePrePostEvaluation(ctx, phaseCtx, workloadVersion, apicommon.PostDeploymentEvaluationCheckType)
 		}
 		return r.PhaseHandler.HandlePhase(ctx,
 			ctxWorkloadTrace,
 			r.getTracer(),
 			workloadVersion,
-			apicommon.PhaseAppPostEvaluation,
+			apicommon.PhaseWorkloadPostEvaluation,
```

Comment:
nice catch 👍 
can we add a regression test here? so we are sure we don't have more of this errors

Reply:
I will create a ticket for that 👍

Reply:
ticket here https://github.com/keptn/lifecycle-toolkit/issues/3121 I will work on it when this PR will get merged